tests/external_executor/external_executor.py

```diff
@@ -481,11 +481,44 @@ def test_logging_executor(network, args):
 
             r = c.post("/app/log/public", {"id": log_id, "msg": log_msg})
             assert r.status_code == 200
-
             r = c.get(f"/app/log/public?id={log_id}")
+
             assert r.status_code == 200
             assert r.body.json()["msg"] == log_msg
 
+            # post to private table
+            r = c.post("/app/log/private", {"id": log_id, "msg": log_msg})
+            assert r.status_code == 200
+            tx_id = r.headers.get("x-ms-ccf-transaction-id")
+
+            # make a historical query
+            timeout = 3
+            start_time = time.time()
+            end_time = start_time + timeout
+            success_msg = ""
+            while time.time() < end_time:
+                headers = {"x-ms-ccf-transaction-id": tx_id}
+                r = c.get(f"/app/log/private/historical?id={log_id}", headers=headers)
+                if r.status_code == http.HTTPStatus.OK:
+                    assert r.body.json()["msg"] == log_msg
+                    success_msg = log_msg
```

Comment:
```suggestion
                    break
```

I think we've asserted equality on the line above, and can end the loop here?

Reply:
Yup, ended the loop. But I didn't remove the `success_msg = log_msg` line as it confirms that the query succeeded and not timed out.

Reply:
Ah, that makes sense. You can also do a `while ... else: raise TimeoutError` loop in Python. The `else` block is only executed when the loop exits because the condition fails (ie - the timeout expired), and will be skipped if the loop is executed by a `break`.